src/main/java/org/jenkinsci/plugins/matrixauth/integrations/casc/MatrixAuthorizationStrategyConfigurator.java

```diff
@@ -30,20 +37,80 @@ public Class<?> getImplementedAPI() {
                         .getter(unused -> null)
                         .setter(MatrixAuthorizationStrategyConfigurator::setPermissionsDeprecated)));
     }
 
+    private static class PermissionAssignment {
+        private final PermissionDefinition permission;
+        private final PermissionEntry entry;
+
+        private PermissionAssignment(PermissionDefinition permission, PermissionEntry entry) {
+            this.permission = permission;
+            this.entry = entry;
+        }
+
+        public PermissionDefinition getPermission() {
+            return permission;
+        }
+
+        public PermissionEntry getEntry() {
+            return entry;
+        }
+    }
+
+    /**
+     * Maps an {@link AuthorizationContainer} to a collection (list) of {@link DefinitionEntry}, its serialized form.
+     *
+     * @param container the container
+     * @return
+     */
+    public static Collection<DefinitionEntry> getEntries(AuthorizationContainer container) {
+        // Contain has: Map from Permission to List of PermissionEntries (sid and type)
+        final Map<Permission, Set<PermissionEntry>> entries = container.getGrantedPermissionEntries();
+
+        final HashMap<PermissionEntry, List<PermissionDefinition>> intermediate = entries.entrySet().stream()
+                .map(entry -> Map.entry(PermissionDefinition.forPermission(entry.getKey()), entry.getValue()))
+                .flatMap(entry -> entry.getValue().stream().map(p -> new PermissionAssignment(entry.getKey(), p)))
+                .collect(
+                        HashMap::new,
+                        (c, e) -> c.computeIfAbsent(e.getEntry(), f -> new ArrayList<>())
+                                .add(e.getPermission()),
+                        (c1, c2) -> {
+                            /* unused */
+                        });
+        final List<DefinitionEntry> result = intermediate.entrySet().stream()
+                .map(entry -> new DefinitionEntry(
+                        entry.getKey().getType(),
+                        new DefinitionEntry.Child(entry.getKey().getSid(), entry.getValue())))
+                .sorted(Comparator.comparing(DefinitionEntry::permissionEntry))
+                .collect(Collectors.toList());
+        return result;
+    }
+
+    public static void setEntries(AuthorizationContainer container, Collection<DefinitionEntry> entries) {
+        entries.forEach(e -> {
+            e.child().getPermissions().stream()
+                    .map(PermissionDefinition::getPermission)
+                    .forEach(p -> {
+                        container.add(p, e.permissionEntry());
+                    });
+        });
+    }
+
     /**
      * Extract container's permissions as a List of "TYPE:PERMISSION:sid"
      */
-    public static Collection<String> getPermissions(AuthorizationContainer container) {
+    public static Collection<String> getLegacyPermissions(AuthorizationContainer container) {
```

Comment:
method appears to be unused

Reply:
Thanks, filed https://github.com/jenkinsci/matrix-auth-plugin/pull/153 and will clean that up in the next release.